packages/integrations/components/integrationsManager/index.js

```diff
@@ -0,0 +1,144 @@
+import { actionHydrateComponents } from '../../actions';
+// Integration components.
+import GoogleAnalytics from '../googleAnalytics';
+
+/**
+ * Integrations Manager.
+ *
+ * A component that consumes a `irving/integrations` configuration from the
+ * REST API, Irving config, or persisted Redux state slice and renders the
+ * corresponding integrations on the client.
+ *
+ * @param {Object}   props - All props.
+ * @param {Object}   props.integrations - Config object for active integrations.
+ * @param {boolean}  props.isHydrated - Has the component map already been hydrated?
+ * @param {function} props.hydrateComponents - Redux action for updating the `componentMap` in the state tree.
+ * @param {Object}   props.hydratedComponents - The existing component key/prop pairs.
+ */
+const IntegrationsManager = ({
+  integrations,
+  isHydrated,
+  hydrateComponents,
+  hydratedComponents,
+}) => {
+  // Declare a map of available integration components with a corresponding key.
+  const componentMap = [
+    { key: 'googleAnalytics', el: GoogleAnalytics },
+  ];
+  // This check is run on server-side renders to ensure that the component hydration
+  // is ran a single time. After hydration, `isHydrated` is set to `true` and will
+  // only be re-run if the component props change during the component's lifecycle.
+  if (false === isHydrated) {
+    // Retrieve the keys available for render in the component map.
+    const keyMap = Object.keys(integrations);
+
+    // Create an array for hydrated component keys and props.
+    const components = [];
+
+    componentMap.forEach((component, index) => {
+      // Get the component's key.
+      const { key } = component;
+
+      if (- 1 < keyMap.indexOf(key)) {
+        // Retrieve the key/value pairs set for the integration and convert them into props.
+        // If a user config override exists, use it instead.
+        let props;
+        if ('integrations' in userConfig && key in userConfig.integrations) {
+          props = userConfig.integrations[key];
+        } else {
+          props = integrations[key];
+        }
+        // Update the item at the current index with the transformed component.
+        components[index] = { key, props };
+      }
+    });
+
+    // Update the components in the store.
+    hydrateComponents(components);
+  }
+
+  /**
+   * A function that takes a component key/props pair, matches the key with a
+   * component that exists in the `componentMap`, and returns a valid React
+   * element with the associated key and props applied.
+   *
+   * @param {Object} component - The component key and props.
+   */
+  const renderComponents = (component) => {
+    // Create a map of available component keys.
+    const keyMap = componentMap.map((co) => co.key);
+    const index = keyMap.indexOf(component.key);
+
+    // If the current component key exists in the key map, clone that component
+    // with the hydrated props for render.
+    if (- 1 < index) {
```

Comment:
Could use `keyMap.includes(component.key)` here too

Reply:
True, but since we need the index value for the retrieving the component for the clone, the use here makes sense IMO

Reply:
👍